lib/internal/modules/esm/loader.js

```diff
@@ -319,6 +336,15 @@ class DefaultModuleLoader {
     return result;
   }
 
+  loadSync(url, context) {
+    defaultLoadSync ??= require('internal/modules/esm/load').defaultLoadSync;
+
+    const result = defaultLoadSync(url, context);
+    result.format = `require-${result?.format}`;
```

Comment:
This feels awkward. Could we perhaps track the origin of each load as a distinct property, like whether the load originated as a `require` or an `import`?

Reply:
Wouldn't `import` necessarily have an async load?

Reply:
The `?.` is extraneous, though. Either this is returning a null result, or it isn't. If it is, then assigning to `result.format` will fail.